Slurm support

t-ramz opened the pull request:
I have a user at ORNL's OLCF that has requested to use sold_dmft on one of our resources and I have had trouble making it run with the slurm workload manager. I made a few changes here that can help alleviate the problem, and make it "work," but there remains a problem I'm having where submitted sub-jobs are causing an MPI error where it causes an `MPI_ABORT` that may be beyond the scope of this software.

Overall, I think if you want to include slurm-only support out of the box there may need to be a switch or feature in place to divvy up tasks from the host MPI process and allocate to VASP/`dft_exe` accordingly. E.g. get all ranks up front running `solid_dmft` and when running `dft_exe` provide explicit rank mappings in some way.
Take this with a grain of salt, I don't know your codebase.

Let me know if you have thoughts or questions and I'll do my best to get back to you!

Commit:
add minimal config for slurm support on OLCF CADES Baseline

Commit:
slurm hostfile path fix

Commit:
add walltime to slurm config

Commit:
fix hostnames for baseline

the-hampel commented on python/solid_dmft/dft_managers/mpi_helpers.py:
```diff
@@ -59,6 +59,9 @@ def create_hostfile(number_cores, cluster_name):
 
     hostnames = mpi.world.gather(socket.gethostname(), root=0)
+    if cluster_name == 'slurm':
+        slurm_hostnames = [hostname.split('.')[0] for hostname in hostnames]  # TODO: please find a better solution
```
can you enlighten me why this extra part is necessary in this case?

the-hampel commented on python/solid_dmft/dft_managers/mpi_helpers.py:
```diff
@@ -148,6 +152,14 @@ def get_mpi_arguments(mpi_profile, mpi_exe, number_cores, dft_exe, hostfile):
 
+    if mpi_profile == 'slurm':
+        return [
+            mpi_exe, '-n', str(number_cores), '--export=PATH',
```
should we enforce here that mpi_exe is 'srun' ?

the-hampel commented:
Hi @t-ramz ,
 thank you for your addition! We can of course add such option. See two questions / comments I have above. 

Regarding MPI_ABORT. If you have specific ideas let me know. We currently have anyway trouble often. For example if one of the ranks sends a signal 9 / 15 or something often other ranks don't die etc. 

Best,
Alex

the-hampel committed:
[dft_managers] clean up SLURM mpi_helpers TODOs

- Remove TODO in create_hostfile: stripping the domain suffix with
  split('.')[0] is the correct way to obtain short hostnames for the
  SLURM nodefile (equivalent to `hostname -s`)
- Default mpi_exe to 'srun' for mpi_profile='slurm'; the global default
  'mpirun' is overridden automatically, but any explicit dft.mpi_exe
  setting in the config is respected
- Remove hardcoded '-t 05:00' time limit: srun within an active SLURM
  allocation inherits the remaining job time automatically when -t is
  omitted, so no explicit limit is needed
